D/D.sublime-syntax

```diff
@@ -8,339 +8,1971 @@ file_extensions:
+          set: [version-condition-end, integer]
+        - include: not-whitespace-illegal
+    - include: not-whitespace-illegal
+  version-condition-end:
+    - match: '\)'
+      scope: punctuation.section.brackets.end.d
+      pop: true
+    - include: not-whitespace-illegal
+  debug-condition:
+    - match: '\('
+      scope: punctuation.section.brackets.begin.d
+      set:
+        - match: '\b({{name}})\b'
+          scope: constant.other.d
+          set: version-condition-end
+        - match: '{{integer_lookahead}}'
+          set: [version-condition-end, integer]
+        - include: not-whitespace-illegal
+    - match: '(?=\S)'
+      pop: true
+  static-foreach-in:
+    - match: '\b(static)\s+(foreach|foreach_reverse)\b'
       captures:
-        1: punctuation.definition.comment.d
-  constant_placeholder:
-    - match: '(?i:%(\([a-z_]+\))?#?0?\-?[ ]?\+?([0-9]*|\*)(\.([0-9]*|\*))?[hL]?[a-z%])'
-      scope: constant.other.placeholder.d
-  function_assertions:
-    - match: '(in|out)\s*(?=\{)'
+        1: keyword.control.flow.d
+        2: keyword.control.flow.d
+      push: [optional-block-statement, foreach-arguments]
+  static-assert-in:
+    - match: '\b(static)\s+(assert)\b'
+      captures:
+        1: keyword.control.conditional.d
+        2: keyword.control.conditional.d
+      push: [expect-end-of-line, assert-arguments]
+  constructions-in:
+    - match: '{{construction_lookahead}}'
+      push: construction
+  construction:
+    # Post-blit constructor
+    - match: '\b(this)\s*(\()\s*(this)\s*(\))'
+      captures:
+        1: entity.name.function.post-blit.d
+        2: punctuation.section.group.begin.d
+        3: variable.language.d
+        4: punctuation.section.group.end.d
+      set: function-definition-after-arguments
+    - match: '\b(this)\s*(?=\(|$)'
       captures:
-        1: keyword.control.statement.d
+        1: entity.name.function.constructor.d
+        2: punctuation.section.group.begin.d
+      set: constructor-argument-list
+    - match: '(~\s*this)\s*(\()\s*(\))'
+      captures:
+        1: entity.name.function.destructor.d
+        2: punctuation.section.group.begin.d
+        3: punctuation.section.group.end.d
+      set: function-definition-after-arguments
+    - include: not-whitespace-illegal-pop
+  # This could either be function/template parameters or arguments. Assume
+  # arguments and switch to parameters when possible
+  constructor-argument-list:
+    - match: '\('
+      scope: punctuation.section.group.begin.d
+      set:
+        - match: '\)'
+          scope: punctuation.section.group.end.d
+          set: function-definition-after-first-argument-list
+        - match: '{{parameter_attribute_lookahead}}'
+          set: [function-definition-after-first-argument-list, function-argument-definition-list-content]
+        - match: '(?=\S)'
+          set: [constructor-argument-list-after-value, value]
+    - include: not-whitespace-illegal-pop
+  constructor-argument-li
```

Comment:
Hint: indention too low.

Reply:
I don't quite follow, sorry. What do you mean by this?

Reply:
Just see, picked the wrong lines: Please check `invariant-block` context. SublimeLint commplained about the rules after `set` to be inproperly indented.

```
    - match: '{'
      scope: punctuation.section.block.begin.d
      set:
      - match: '}'
        scope: punctuation.section.block.end.d
        pop: true
      - include: statement-list-in
```

Reply:
SublimeLint looks like a framework for other linters to me. I'm not sure what linter you're running on this. If its complaining about the `- include: not-whitespace-illegal-pop` line then its certainly a false positive. The contents are handled by `(?=\S)` and `not-whitespace-illegal-pop` just highlights everything as invalid.

Reply:
Nope: Like I said, I selected the wrong lines by accident. It's not `- include: not-whitespace-illegal-pop` the linter complains about but the construct of my previous post in line 1663ff.

Sorry for the confustion.

Reply:
Ah yes, sorry for not catching on sooner.